app/controllers/etds_controller.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+require 'resolv'
+
+# Controller for ETD endpoints used by Peoplesoft to transfer data to the ETD system
+class EtdsController < ApplicationController
+  skip_verify_authorized only: %i[index create]
+  before_action :authenticate, only: %i[index create], unless: proc { request_from_authorized_origin? }
```

Comment:
Given that index and create are the only two actions in this controller, we can prolly ditch the `only` stuff